yapio.c

```diff
@@ -228,6 +228,7 @@ typedef struct yapio_blk_metadata
     int    ybm_writer_rank;     //rank who has last written this block
     int    ybm_owner_rank_fpp;  //fpp only, file to which contents belong
     size_t ybm_blk_number;      //number of the block - should not change!
+    bool   ybm_not_hole;        //block is empty
```

Comment:
Adding a new member here increases the structure size by 8 bytes - which is considerable for this structure (since there's one per block).  Can you please figure out a way to squeeze this into a single bit possibly by using a bitfield?  ybm_blk_number should be fine with only 63 bits.